cmd/share-secret/main.go

```diff
@@ -0,0 +1,69 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/gorilla/mux"
+)
+
+func main() {
+	if err := run(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func run() error {
+
+	r := mux.NewRouter()
+	r.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		if _, err := fmt.Fprintln(w, "hello world"); err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+	})
+
+	r.Handle("/secret", Apply(PostSecret(), CORSWithDefaults())).Methods(http.MethodPost)
+	r.HandleFunc("/secret/{key}", sharesecretlink).Methods(http.MethodGet)
+
+	log.Fatal(http.ListenAndServe(":8080", r))
```

Comment:
Because this function already returns an error if it fails, why not `return http.ListenAndServe(":8080", r)` without the `log` part?
If `err` is not nil, `main()` will `log.Fatal()` anyway